versioned/tiered/rocksdb/src/main/java/org/projectnessie/versioned/rocksdb/EntityConverter.java

```diff
@@ -0,0 +1,172 @@
+          break;
+        case RocksRef.COMMITS_OLD_ID:
+          builder.setOldId(level0.getValue().getBinary());
+          break;
+        case RocksRef.COMMITS_NEW_ID:
+          builder.setNewId(level0.getValue().getBinary());
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * This converts an Entity into KeyMutation protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.KeyMutation entityToKeyMutation(Entity entity) {
+    if (!entity.getType().equals(Entity.EntityType.MAP)) {
+      throw new UnsupportedOperationException();
+    }
+
+    ValueProtos.KeyMutation.Builder builder = ValueProtos.KeyMutation.newBuilder();
+
+    for (Map.Entry<String, Entity> level0 : entity.getMap().entrySet()) {
+      switch (level0.getKey()) {
+        case RocksRef.COMMITS_KEY_ADDITION:
+          builder.setType(ValueProtos.KeyMutation.MutationType.ADDITION);
+          builder.setKey(entityToKey(level0.getValue()));
+          break;
+        case RocksRef.COMMITS_KEY_REMOVAL:
+          builder.setType(ValueProtos.KeyMutation.MutationType.REMOVAL);
+          builder.setKey(entityToKey(level0.getValue()));
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    return builder.build();
+  }
+
+  /**
+   * This converts an Entity into Key protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.Key entityToKey(Entity entity) {
+    if (!entity.getType().equals(Entity.EntityType.LIST)) {
+      throw new UnsupportedOperationException();
+    }
+
+    return ValueProtos.Key.newBuilder().addAllElements(
+      entity.getList().stream().map(Entity::getString).collect(Collectors.toList())).build();
+  }
+
+  /**
+   * This converts an Entity into KeyDelta protobuf object.
+   * @param entity the entity to convert
+   * @return the object converted from the entity
+   */
+  public static ValueProtos.KeyDelta entityToKeyDelta(Entity entity) {
+    if (entity.getType() != Entity.EntityType.MAP) {
+      throw new UnsupportedOperationException("Invalid value for keyDelta");
+    }
+
+    final Map<String, Entity> entityMap = entity.getMap();
+    final ValueProtos.KeyDelta.Builder keyDeltaBuilder = ValueProtos.KeyDelta.newBuilder();
+
+    for (Map.Entry<String, Entity> entry : entityMap.entrySet()) {
+      switch (entry.getKey()) {
+        case RocksL3.TREE_ID:
+          keyDeltaBuilder.setId(entry.getValue().getBinary());
+          break;
+        case RocksL3.TREE_KEY:
+          keyDeltaBuilder.setKey(entityToKey(entry.getValue()));
+          break;
+        default:
+          throw new UnsupportedOperationException(String.format("Unknown field \"%s\" for keyDelta", entry.getKey()));
```

Comment:
Why is this the only exception which has an error message in this class?

Reply:
Added messages to the other exceptions.